aten/src/ATen/native/Itertools.cpp

```diff
@@ -0,0 +1,60 @@
+#include "ATen/ATen.h"
+#include "ATen/Dispatch.h"
+
+#include <vector>
+
+namespace {
+
+using namespace at;
+
+Tensor _triu_mask(int64_t n, int64_t dims, bool diagonal, TensorOptions opt) {
+  // get a mask that has value 1 whose indices satisfies i < j < k < ...
+  // or i <= j <= k <= ... (depending on diagonal)
+  Tensor range = at::arange(n, opt.dtype(kLong));
+  std::vector<Tensor> index_grids = at::meshgrid(std::vector<Tensor>(dims, range));
+  Tensor mask = at::ones(index_grids[0].sizes(), opt.dtype(kByte));
+  if(diagonal) {
+    for(int64_t i = 0; i < dims - 1; i++) {
+      mask *= index_grids[i] <= index_grids[i+1];
+    }
+  } else {
+    for(int64_t i = 0; i < dims - 1; i++) {
+      mask *= index_grids[i] < index_grids[i+1];
+    }
+  }
+  return mask;
+}
+
+}  // namespace
+
+namespace at {
+namespace native{
+
+Tensor cartesian_prod(TensorList tensors) {
+  for(const Tensor &t : tensors) {
+    AT_CHECK(t.dim() == 1, "Expect a 1D vector, but got shape ", t);
```

Comment:
Should be `t.size()` or `t.sizes()` (don't remember which one), right  now the error message would print out the tensor which is undesirable especially if the tensor is large.